### PR TITLE
Automl metric shape recovery & Forecaster multioutput option

### PR DESCRIPTION
**Pull request by TheaperDeng**
Mainly three changes for this PR
1. Make sure that the metric evaluation result shape is the same as input shape without the first dim. 
2. Add multioutput option for Zouwu Forecaster(TCN and Seq2seq)
3. Change deprecated API in automl metric test. (related to issue #3307)

### CODE REVIEW

File: docs/docs/Zouwu/API/TCNForecaster.md [hunk 1]
```diff
@@ -50,23 +50,24 @@ fit(x, y, validation_data=None, epochs=1, metric="mse", batch_size=32)
...
 ```python
-evaluate(x, y, metrics=['mse'])
+evaluate(x, y, metrics=['mse'], multioutput="uniform_average")
 ```
 
 - **`x`**: A numpy array with shape (num_samples, lookback, feature_dim).
 - **`y`**: A numpy array with shape (num_samples, horizon, target_dim).
 - **`metrics`**: A list contains metrics for test/valid data.
-
+- **`multioutput`**: Defines aggregating of multiple output values. String in ['raw_values', 'uniform_average']
```
Comment (yushan111) [inline docs/docs/Zouwu/API/TCNForecaster.md:59]: default is

File: pyzoo/zoo/automl/common/metrics.py [hunk 2]
```diff
@@ -238,7 +240,9 @@ def MSE(y_true, y_pred, multioutput='uniform_average'):
         A non-negative floating point value (the best value is 0.0), or an
         array of floating point values, one for each individual target.
     """
-    y_true, y_pred = _standardize_input(y_true, y_pred, multioutput)
+    y_true, y_pred, original_shape = _standardize_input(y_true, y_pred, multioutput)
+    if multioutput == 'raw_values':
+        return mean_squared_error(y_true, y_pred, multioutput=multioutput).reshape(original_shape)
     return mean_squared_error(y_true, y_pred, multioutput=multioutput)
```
Comment (yushan111) [inline pyzoo/zoo/automl/common/metrics.py:246]: the code here might be a little duplicated
```python
result = mean_squared_error(y_true, y_pred, multioutput=multioutput)
if multioutput == 'raw_values':
     return result.reshape(original_shape)
return result

File: pyzoo/zoo/automl/model/base_pytorch_model.py [hunk 3]
```diff
@@ -164,13 +164,13 @@ def _print_model(self):
         for i in range(len(list(self.model.parameters()))):
             print(list(self.model.parameters())[i].size())
 
-    def evaluate(self, x, y, metrics=['mse']):
+    def evaluate(self, x, y, metrics=['mse'], multioutput="raw_values"):
```
Comment (yushan111) [inline pyzoo/zoo/automl/model/base_pytorch_model.py:167]: change default to mean?

File: pyzoo/zoo/zouwu/model/forecast/tcn_forecaster.py [hunk 4]
```diff
@@ -131,19 +131,24 @@ def predict_with_onnx(self, x, dirname=None):
...
-    def evaluate(self, x, y, metrics=['mse']):
+    def evaluate(self, x, y, metrics=['mse'], multioutput="uniform_average"):
         """
         Evaluate using a trained forecaster.
 
         :param x: A numpy array with shape (num_samples, lookback, feature_dim).
         :param y: A numpy array with shape (num_samples, horizon, target_dim).
         :param metrics: A list contains metrics for test/valid data.
+        :param multioutput: Defines aggregating of multiple output values.
+               String in ['raw_values', 'uniform_average']
```
Comment (yushan111) [inline pyzoo/zoo/zouwu/model/forecast/tcn_forecaster.py:142]: defaults to

**Review by yushan111 — COMMENTED** [review]
Others LGTM

**Comment by TheaperDeng** [comment]
PR validation combine pass:
http://10.239.47.210:18888/job/ZOO-PR-Validation/5397/
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests/2637/
http://10.239.47.210:18888/job/ZOO-PR-Scala-Spark-3.0/829/

**Pull request merged**